src/explore-education-statistics-admin/src/pages/release/edit-release/content/ReleaseContext.tsx

```diff
@@ -0,0 +1,237 @@
+import produce from 'immer';
+import {
+  EditableContentBlock,
+  ExtendedComment,
+  EditableRelease,
+} from '@admin/services/publicationService';
+import { DataBlock } from '@common/services/dataBlockService';
+import { ContentSection } from '@common/services/publicationService';
+import React, { createContext, useContext, useReducer } from 'react';
+import ReleaseDispatchAction from './actions';
+
+type Dispatch = (action: ReleaseDispatchAction) => void;
+type State = {
+  release: EditableRelease | undefined;
+  canUpdateRelease: boolean;
+  availableDataBlocks: DataBlock[];
+  unresolvedComments: ExtendedComment[];
+};
+type ReleaseProviderProps = { children: React.ReactNode };
+
+const ReleaseStateContext = createContext<State | undefined>(undefined);
+const ReleaseDispatchContext = createContext<Dispatch | undefined>(undefined);
+
+export function releaseReducer(state: State, action: ReleaseDispatchAction) {
+  switch (action.type) {
+    case 'CLEAR_STATE':
+      return {
+        release: undefined,
+        canUpdateRelease: false,
+        availableDataBlocks: [],
+        unresolvedComments: [],
+      };
+    case 'SET_STATE':
+    case 'SET_AVAILABLE_DATABLOCKS':
+      return { ...state, ...action.payload };
+    case 'REMOVE_BLOCK_FROM_SECTION':
+      return produce<State>(state, draft => {
+        const { sectionId, blockId, sectionKey } = action.payload.meta;
+        if (
+          draft.release === undefined ||
+          typeof draft.release[sectionKey] !== 'object'
+        ) {
+          throw new Error('REMOVE_BLOCK_FROM_SECTION: failed');
+        } else if (sectionKey === 'content') {
```

Comment:
The above conditional throws an error so it would be nicer to just write this `else if` as a new `if` to break it up a bit